components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -4361,20 +4272,11 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attributionId '${attributionId}'`);
         }
 
-        let team: Team | undefined;
-        if (attrId.kind === "team") {
-            team = (await this.guardTeamOperation(attrId.teamId, "update", "not_implemented")).team;
-        } else {
-            if (attrId.userId !== user.id) {
-                throw new ResponseError(
-                    ErrorCodes.PERMISSION_DENIED,
-                    "Cannot create Stripe customer profile for another user",
-                );
-            }
-        }
+        const team = (await this.guardTeamOperation(attrId.teamId, "update", "not_implemented")).team;
 
+        //TODO billing email should be editable within the org
         const billingEmail = User.getPrimaryEmail(user);
-        const billingName = attrId.kind === "team" ? team!.name : User.getName(user);
+        const billingName = team!.name;
```

Comment:
is the `!` needed here? And if so, do we need to handle when `team` is undefined?

Reply:
it is not needed